cunumeric/array.py

```diff
@@ -1252,6 +1252,92 @@ def dump(self, file):
     def dumps(self):
         return self.__array__().dumps()
 
+    def fft(self, s, axes, kind, direction, norm):
+        # Dimensions check
+        if self.ndim > 3:
+            raise NotImplementedError(
+                f"{self.ndim}-D arrays are not supported yet"
+            )
+
+        # Type
+        fft_output_type = self.dtype
+        if kind == FFTCode.FFT_D2Z:
+            fft_output_type = np.complex128
+        elif kind == FFTCode.FFT_R2C:
+            fft_output_type = np.complex64
+        elif kind == FFTCode.FFT_Z2D:
+            fft_output_type = np.float64
+        elif kind == FFTCode.FFT_C2R:
+            fft_output_type = np.float32
+
+        # Axes and sizes
+        user_axes  = axes is not None
+        user_sizes = s is not None
+        if not user_axes:
+            fft_axes = list(range(-len(s),0)) if user_sizes else list(range(self.ndim))
+        else:
+            fft_axes = axes
+        fft_axes = [x % self.ndim for x in fft_axes]
```

Comment:
Why `%` ? Shouldn't we error out if `fft_axis = [5]` but `self.ndim = 3` ?

Reply:
Good catch! I do the modulo to convert negative axes, but we need to add a check for dimensions

Reply:
Ha I understand why the modulo now; I thought the intent was really to map 5 to 2 if `ndim == 3` for instance. But a check would be good anyway.